internal/asm/f64: incorrect value returned by L2NormInc for zero-length vector

I am seeing failure in `L2NormInc` on travis, but only on macOS and only for go1.13, and not always. This has happened 3 times that I've seen. The value that's being returned is the expected value for the last test‽

```
--- FAIL: TestL2NormInc (0.00s)
    l2norm_test.go:70: Test 0 L2NormInc error Got: 2.236068 Expected: 0.000000
```

We don't run `noasm`-tagged tests on macOS, so this is presumably something happening in the assembly.


/cc @Kunde21

--- a/internal/asm/f64/l2norminc_amd64.s
+++ b/internal/asm/f64/l2norminc_amd64.s
@@ -36,10 +36,10 @@ TEXT ·L2NormInc(SB), NOSPLIT, $0
 	MOVQ n+24(FP), LEN    // LEN = len(x)
 	MOVQ incX+32(FP), INC
 	MOVQ x_base+0(FP), X_
+	XORPS ZERO, ZERO
 	CMPQ LEN, $0          // if LEN == 0 { return 0 }
 	JZ   retZero
 
-	XORPS ZERO, ZERO
 	XORPS INFMASK, INFMASK
 	XORPS NANMASK, NANMASK
 	MOVSD $1.0, SUMSQ           // ssq = 1

